dspace-server-webapp/src/test/java/org/dspace/app/rest/HealthIndicatorsIT.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import static org.dspace.app.rest.link.search.HealthIndicatorMatcher.match;
+import static org.dspace.app.rest.link.search.HealthIndicatorMatcher.matchDb;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.Map;
+
+import org.dspace.app.rest.test.AbstractControllerIntegrationTest;
+import org.junit.Test;
+import org.springframework.boot.actuate.health.Status;
+
+/**
+ * Integration tests to verify the health indicators configuration.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ *
+ */
+public class HealthIndicatorsIT extends AbstractControllerIntegrationTest {
+
+    private static final String HEALTH_PATH = "/actuator/health";
+
+    @Test
+    public void testWithAnonymousUser() throws Exception {
+
+        getClient().perform(get(HEALTH_PATH))
+            .andExpect(status().isServiceUnavailable())
+            .andExpect(jsonPath("$.status", is(Status.OUT_OF_SERVICE.getCode())))
+            .andExpect(jsonPath("$.components").doesNotExist());
+
+    }
+
+    @Test
+    public void testWithNotAdminUser() throws Exception {
+
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        getClient(token).perform(get(HEALTH_PATH))
+            .andExpect(status().isServiceUnavailable())
+            .andExpect(jsonPath("$.status", is(Status.OUT_OF_SERVICE.getCode())))
+            .andExpect(jsonPath("$.components").doesNotExist());
+
+    }
+
+    @Test
+    public void testWithAdminUser() throws Exception {
+
+        String token = getAuthToken(admin.getEmail(), password);
+
+        getClient(token).perform(get(HEALTH_PATH))
+            .andExpect(status().isServiceUnavailable())
+            .andExpect(jsonPath("$.status", is(Status.OUT_OF_SERVICE.getCode())))
```

Comment:
It seems a bit odd to have the entire `/health` actuator display "OUT OF SERVICE" if only the `geoIp` check fails.  I'd consider `geoIp` to be optional, as not every site really uses Solr Statistics...and the site will function perfectly well even if `geoIP` is missing.

So, in this test, I can understand having the `geoIp` component respond with "OUT OF SERVICE".  But it seems odd for the entire status to then become "OUT OF SERVICE".